src/data/fin.lean

```diff
@@ -504,6 +522,17 @@ order_embedding.of_strict_mono (λ a, cast_lt a (lt_of_lt_of_le a.2 h)) $ λ a b
 @[simp] lemma cast_le_mk (i n m : ℕ) (hn : i < n) (h : n ≤ m) :
   cast_le h ⟨i, hn⟩ = ⟨i, lt_of_lt_of_le hn h⟩ := rfl
 
+@[simp] lemma range_cast_le {n k : ℕ} (h : n ≤ k) :
+  set.range (cast_le h) = {i | (i : ℕ) < n} :=
+set.ext (λ x, ⟨λ ⟨y, hy⟩, hy ▸ y.2, λ hx, ⟨⟨x, hx⟩, fin.ext rfl⟩⟩)
+
+@[simp] lemma coe_set_range_cast_le_symm {n k : ℕ} (h : n ≤ k) (i : fin k) (hi) :
+  ((equiv.set.range _ ((cast_le h).injective)).symm ⟨i, hi⟩ : ℕ) = i :=
```

Comment:
nit: parens
```suggestion
  ((equiv.set.range _ (cast_le h).injective).symm ⟨i, hi⟩ : ℕ) = i :=
```

Do you actually use this lemma anywhere?

Reply:
I don't think so (nor the one for `cast_succ`), since everything is now `range_of_left_inverse`. Should I just get rid of them because they are too specific?

Reply:
I'd probably be inclined to get rid of at least the `equiv.set.range` lemmas, yes - if `coe` is the inverse, then probably `equiv.set.range` was the wrong definition to use in the first place.

The ones about `set.range` are probably harmless to keep,